core/src/main/java/org/apache/iceberg/encryption/AesGcmInputFile.java

```diff
@@ -26,28 +26,42 @@ public class AesGcmInputFile implements InputFile {
   private final InputFile sourceFile;
   private final byte[] dataKey;
   private final byte[] fileAADPrefix;
-  private long plaintextLength;
+  private Long encryptedLength;
+  private Long plaintextLength;
 
   public AesGcmInputFile(InputFile sourceFile, byte[] dataKey, byte[] fileAADPrefix) {
+    this(sourceFile, dataKey, fileAADPrefix, null);
+  }
+
+  public AesGcmInputFile(InputFile sourceFile, byte[] dataKey, byte[] fileAADPrefix, Long length) {
```

Comment:
Added this so that `StandardEncryptionManager` can pass the length from the key metadata.